### DISCUSSION TITLE
Actions loop on 2 or more of the same identify objects on the same page

### DISCUSSION THREAD

**Discussion by Kuryuuki** [getgauge/taiko#2333] [General]
Hello how are you? I'm pretty new and I'm building a code but I'm pretty locked up.

I would like to know if it is possible to do a loop of actions in sequence of equal objects.

The objects are the same to identify, but they are not the same in practice, each one represents a profile of a user. The page sometimes has 2,3,4... profiles

I'm not able to make Taiko go to the next same object later to perform the actions on the object and I don't know how to create an efficient loop for an indeterminate number

Example 1:
Object 1 = $('div.navigation-icon');
Action code
goto url

Example 2
Object 1 = $('div.navigation-icon');
Action code
Object 2 = $('div.navigation-icon');
Action code
goto url

Example 3
Object 1 = $('div.navigation-icon');
Action code
Object 2 = $('div.navigation-icon');
Action code
Object 3 = $('div.navigation-icon');
Action code
goto url

**Comment by zabil** [comment]
By default Taiko returns the first match. There's documentation with a few examples at https://docs.taiko.dev/working_with_element_lists/  if you want to work with multiple matches and loop through them.

**Reply by Kuryuuki** [reply]
@zabil if can u help me with the code? put to select in loop this **await click ($('div.navigation-icon'));** ?

**Reply by Kuryuuki** [reply]
select first **($('div.navigation-icon'));** after that second **($('div.navigation-icon'));**, ....

**Reply by Kuryuuki** [reply]
when you no longer have the button put to give **goto link;**

**Reply by zabil** [reply]
Something like this

```
var navs = await $('div.navigation-icon').elements(); 
for (nav of navs) {
   // actions, for example
   await click(nav);
}
```

**Reply by Kuryuuki** [reply]
I tried to create a simple code to follow followers of a specific page on twitter but it always gives an error after a certain time running:

> Found too many matches. Please use a selector that is more specific
>     at /home/pc/node_modules/taiko/lib/actions/pageActionChecks.js:187:13
>     at async waitAndGetActionableElement (/home/pc/node_modules/taiko/lib/actions/pageActionChecks.js:156:3)
>     at async click (/home/pc/node_modules/taiko/lib/actions/click.js:57:21)
>     at async module.exports.tap (/home/pc/node_modules/taiko/lib/taiko.js:1492:3)
>     at async module.exports.<computed> (/home/pc/node_modules/taiko/lib/taiko.js:2622:14)
>     at async global.<computed> (/home/pc/node_modules/taiko/bin/runFile.js:38:15)
>     at async /home/pc/Desktop/INT/bots/twitter.js:32:17
>  ✔ Browser closed


My code:
```
const { openBrowser, goto, near, click, textBox, press, write, button, reload, text, tap } = require('taiko');
const random = require('random');
const delay = require('delay');

(async () => {
	try 
	{
		var log = 0;

		if (log == 0) 
		{
			await openBrowser();
			await goto('twitter.com/i/flow/login');
			await write('USER', textBox({ name: 'username' }));
            		await click('Next');
			await write('PASS', textBox({ name: 'password' }));
			await click('Next');
            		await delay(2500);
            		await goto('twitter.com/facebook/followers')
			log = log + 1;
		}

        var time = 0;

		while (true)
		{

            if(await text('Follow', { exactMatch: true}).exists())
            {
                await tap('Follow', { exactMatch: true}); //TEST
                await click('Follow', { exactMatch: true});

                var tex = random.int(1, 3);

                if(tex == 3)
                {
                    await delay(5001);
                }

                if(tex == 2)
                {
                    await delay(10002);
                }

                if(tex == 1)
                {
                    await delay(15003);
                }
            }

            else
            {
                await scrollDown(2500)
            }
		}
	}

	finally 
	{

	}

})();
```

**Reply by sriv** [reply]
the default `noOfElementToMatch` is [set to `20`](https://github.com/getgauge/taiko/blob/68d2715f93e128a40ffba1cb97b48c0dff1da61d/lib/config.js#L6). You can try using [`setConfig`](https://docs.taiko.dev/api/setconfig/) to increase this.
